GUI implementation

rikigigi opened the pull request:
A Graphical User Interface for thermocepstrum.

The main features are implemented and work.
Will require further improvement work (see `GUI_WISHLIST.md`) and testing.

sebaB003 committed:
Created a new branch: GUI
Set some global settings and preferences that will be used by the GUI.
Created some files that will manage the GUI and its functionality.
Created multi-window system for the GUI.
Created the main window.
Created the top-bar.

sebaB003 committed:
Added the file manager window.
Added the file manager tool.
Added some functions to load settings.

sebaB003 committed:
Updated the multy-window system.
Fixed the status-bar.
Embedded matplotlib in tkinter.

sebaB003 committed:
Started to develo the graph widget.

sebaB003 committed:
Updated the graph widget.
Added the f* selector.
Added the logs and info text box.
Updated the file manager.
Added to the file manager an input format selector.

sebaB003 committed:
Updated the graph widget.
Added an entry to select the data.
Added error and warnings messages to the file manager.

rikigigi committed:
Merge branch 'develop' into GUI

rikigigi committed:
first implementation of spectrum plots

sebaB003 committed:
Updated f* selector.
Added p* selector.
Added Data class to store the values that will be reused and to
generate outputs.

sebaB003 committed:
Added some functions to the Graph class.
Updated the structure of the Data class.
Added the support to display multiple graphs in a single canvas.
Updated the P* selector.

Commit:
Added a checkbox list widget.
Added a preview of the file in the file manager.
Added the psd_width (filter) selector.
Removed the dependecy of the main.py from thermocepstrum.

TODO: check factors in plots and in moving average filter.
TODO: check y-range variation bug.

rikigigi committed:
Merge remote-tracking branch 'origin/develop' into GUI

rikigigi committed:
Merge branch 'develop' into GUI

sebaB003 committed:
Added temperature, volume and DT_FS input fields.
Fixed graphs overlap.

rikigigi committed:
GUI: fixed wrong argument in function (filter window width was not passed)

rikigigi committed:
fixed behaviour of moving average filter
TODO: move stuff from Graph to control_unit
TODO: filter_window_width variable should appear only in one place (for example control_unit)

sebaB003 committed:
Updated GUI layout and structure.
Added info viewer.

Fixed editable logs and infos.

sebaB003 committed:
Fixed the bug in resample current.
Various fixing.
Added function "resample_current" in HeatCurrent.

sebaB003 committed:
Added header description system.
Various fixing.
Removed auto option.

rikigigi committed:
added K_PSD as an argument in  HeatCurrent's cepstral_analysis to manually specify P*

rikigigi committed:
Merge branch 'GUI' of https://github.com/lorisercole/thermocepstrum into GUI

sebaB003 committed:
Updated P* selector.
Updated logs function.

rikigigi committed:
forgot to remove a little thing

sebaB003 committed:
Changend global output method.
Various fix.

sebaB003 committed:
Updated GUI.
Updated graph class.
Added new features to the graph class.
Started to change GUI structure.

rikigigi committed:
packaged thermocepstrum-gui. Now it is installed as an executable.

sebaB003 committed:
Updated F* selector.
Added "View" button to zoom on the cut.

Fixed graph resize bug.
Fixed "open with file manager" button in the file manager frame.
Various fix.

lorisercole committed:
Fixed PSD_FILTER_W definition in resample_current. Fixed minor
copy-paste errors with write_log functions.

sebaB003 committed:
Various fix.
Changed some settings in settings.py

rikigigi committed:
modified behaviour of file manager: the system browser starts from the current directory and then the file manager shows its content

sebaB003 committed:
Changed the CheckList widget behaviour.
Changed the FileManager structure.
Added HeaderSelector window.
Added OtherVariable window.
Updated the logs and info show method.
Added view options in the topbar.

sebaB003 committed:
Fixed back button.

sebaB003 committed:
Fixed environtment variables.
Fixed show/hide logs and info.
Updated view button in Fstar selector.
Various fix.

sebaB003 committed:
Fixed the behaviour of the F* selector.

rikigigi committed:
fixed some incompatibilities with python 2.7 and python 3.5
added readme in the tests directory

rikigigi committed:
Merge branch 'develop' into GUI

rikigigi committed:
Merge branch 'develop' into GUI

rikigigi committed:
TopBar was initialized too much times

rikigigi committed:
Tried to achieve automatic resize in the GUI. When the starting size of the window is not specified, the behaviour is terribly wrong: when the main frame is replaced, the window starts to infinitely resize between the old size and the new 'best size' of the Frame. Nevertheless, if the size is specified, the widgets automagically adapts themselves to the window.
TODO: do this also for the plot frames!
TODO: remove commented lines
TODO: remove bg colors, added for debug

I changed most of the layout managers to grid.

rikigigi committed:
Small cleanups in GUI
The layout can be better.

sebaB003 committed:
Restructured the code in multiple file.
Added some comments.
Changed the layout of some frames.
Updated the graphic.

rikigigi committed:
fixed some forgotten links. The logs are automatically scrolled down.
Variables inside cu.Data are no more class variables, but of the single istance.
The behaviour of the program is wrong when from the fstar selector you go back and then to the next frame again. data.keys is empty and nothing happens.

lorisercole committed:
Merge branch 'develop' into GUI

sebaB003 committed:
Updated graphic.
Updated layout.
Changed layout in P* selector and F* selector.

Fixed log and info show problem in P* selector.

To fix: ckeys are not maintained when go back from F* selector.

sebaB003 committed:
Merge branch 'GUI' of https://github.com/lorisercole/thermocepstrum into GUI

sebaB003 committed:
Added the windows to the topbar.
Added the software icon.
Changed the layout.

sebaB003 committed:
Fixed wrong version visualization in the version window.

rikigigi committed:
packed icons.
setup.json is moved inside thermocepstrum folder, and additional metadata (like version, ecc) is read from there.

rikigigi committed:
Merge branch 'GUI' of https://github.com/lorisercole/thermocepstrum into GUI

sebaB003 committed:
Added multi language support.
Added translation file.
Added italian translation.
Added web view in about window.

Fixed version visualization error.

rikigigi committed:
packaged README.md
a copy is created during the setuptools run, so readme is accessible from the installed package
update missing dependencies

rikigigi committed:
added selector for units of heat current
small beautyfications to the code

TODO: prettify

sebaB003 committed:
Changed the layout of the unit selector in headerSelector.

sebaB003 committed:
Fixed issue where data.keys was deleted

rikigigi committed:
removed redundant data, added symlink

rikigigi committed:
small modifications to texts in the GUI.
TODO: formatting inside the GUI

rikigigi committed:
Extended documentation

sebaB003 committed:
Changed the behaviour of the data.
Add new functionality.

To fix: when "resample" is cliked and when the slider is moved
there are multiple callbacks to the function that resample the graph.

sebaB003 committed:
Updated the layout of the GUI.
Fixed the behaviour of the rasample button.

sebaB003 committed:
Changed the behaviour of the "Next" and "Back" buttons.

rikigigi committed:
bug fix: columnss with "None" were passed

sebaB003 committed:
Updated the behaviour og the GUI.

sebaB003 committed:
Updated the deps.

sebaB003 committed:
Bug fixes and GUI update.
To fix: some times the aic_Kmin is None.

rikigigi committed:
ooops... garbage character in MANIFEST.in

sebaB003 committed:
Fixed the P* selector.

rikigigi committed:
added units in the labels

sebaB003 committed:
Added a new error message when the the software is unable to read
the file.

sebaB003 committed:
Merge branch 'GUI' of https://github.com/lorisercole/thermocepstrum into GUI

rikigigi committed:
implemented loading of npy file, and save of the data in a npy file

rikigigi committed:
when the data is exported, the program saves in the numpy file T, vol and DT_FS. It automatically load them when the file is opened again

TODO: fix bug on last step (when the lo is activated/deactivated there is something wrong; when you recompute the cepstral analysis there is something wrong)

sebaB003 committed:
Moved the logs in a new window.

sebaB003 committed:
Fixed the behaviour of the P* selector.

rikigigi committed:
forgot to remove _var
+ yapf code layout modifications

rikigigi committed:
fixed log problem

rikigigi committed:
Typo in file size

rikigigi committed:
merge

rikigigi committed:
fixed GUI scroll and layout. ScrollFrame *MANUALLY* sets the size of the inner Frame, chosing if it has to expand to the size of the canvas or to left as is, and the user will have to use the scroll bar.

rikigigi committed:
save all config into .npy output file, and load them.
TODO: fix bug in pstarSelector, and set also PSTAR from input file, if present

sebaB003 committed:
Added the settings window.
Added more translations.
Renamed load_path function to load_settings and changed its behaviour.

rikigigi committed:
fixed bug in pstarSelector

rikigigi committed:
Merge branch 'GUI' of https://github.com/lorisercole/thermocepstrum into GUI

sebaB003 committed:
Now you can open only a window per type.

rikigigi committed:
forgot to convert to int settings that are read from file
added Help
moved many strings to languages.json

rikigigi committed:
oopps logic of changes was wrong again

rikigigi committed:
small documentation for the GUI updated

rikigigi committed:
fix rescale of last graph (?)

lorisercole committed:
Updated dependencies and added GUI wishlist

rikigigi committed:
fixed bug of 'none' headers that disappear
fixed bug of multiple DT/V/T selection

rikigigi committed:
fixed bug: wrong plot limits and gui state when the user goes back resetting the variables after leaving the fstar selector frame with the zoom-in option on.

rikigigi committed:
solve #10

rikigigi committed:
added handling of trajectory with odd number of timesteps

rikigigi committed:
separated the library and the command line interface from the GUI in two different python packages. The library+cmd have less dependencies, and they work also with python 2.7.
Small bug fixes in GUI (halt the use if fstar==0)

rikigigi committed:
Small bug fixes in GUI (halt the use if fstar==0)

Script analysis was not loading npy files correctly

rikigigi committed:
oops

rikigigi committed:
script analysis was not loading npy files correctly

rikigigi committed:
update README and some paths

sebaB003 committed:
Added comments in main.py, custom_widgets.py and utils.py

sebaB003 committed:
Added comments in main.py, custom_widgets.py and utils.py

lorisercole committed:
new setup: install GUI only if Python 3

lorisercole commented:
I am going to restore the previous folder structure with a single setup, to avoid splitting the code in 2 packages.
If using Python 2, the GUI will not be installed, guaranteeing maximum compatibility.
Notice that when pip installing, the code will actually be installed in the folders `thermocepstrum` and `thermocepstrum_gui`. Non-code files that should be copied in them are listed in `MANIFEST.in`.

The GUI needs to read the package metadata, contained in `setup.json`. Since this file is not copied during installation, I will make sure a new file `metadata.json` is created in thermocepstrum by the setup code.

lorisercole committed:
Rewrite GUI branch with reorganize_package branch

The original package folder organization has been restored.

lorisercole committed:
Merge branch 'reorganize_package' into GUI

lorisercole committed:
GUI reformatted with yapf, style updated

yapf style updated to avoid single-line arguments of functions with many
arguments.

lorisercole committed:
Update GUI dependencies, bump version

lorisercole reviewed (approved):
First beta version of the **thermocesptrum_GUI**.
All good!

rikigigi committed:
typo
